airflow/providers/amazon/aws/utils/boto.py

```diff
@@ -0,0 +1,65 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""
+This module contains different `boto3` / `botocore` helpers for internal use within the Amazon provider.
```

Comment:
Then lets also add `_` for all functions names so it would be clear for developers using this module

Reply:
Is it really necessary? Because by the almost the same way we exclude from the core almost all `airflow.utils` and only keep as public part `airflow.utils.state`. And for this one as well as for the other internal Amazon providers utils we would not generate API reference for that

![image](https://github.com/apache/airflow/assets/3998685/2083caf2-fd7f-4f7d-8386-f4e90b5733b4)

Reply:
But does it hurt? I quite like using the under score to really indicate things are private. But I know there is some push back for this.

Reply:
Personally when I code I don't read the module description most of the times Python has a convention of telling developers "this is private! Using it is at your own risk" so why not set it? It can only help. I dont see any downside